include/aspect/material_model/interface.h

```diff
@@ -564,6 +590,39 @@ namespace aspect
     }
 
 
+    /**
+     * Some material and heating models need more than just the basic material
+     * model inputs defined in the MaterialModel::MaterialModelInputs
+     * class. These additions are either for more complicated physics
+     * than the basic flow model usually solved by ASPECT (for example
+     * to support the melt migration functionality), or other derived
+     * quantities.
+     *
+     * Rather than litter the MaterialModelInputs class with additional
+     * fields that are not universally used, we use a mechanism by
+     * which MaterialModelInputs can store a set of pointers to
+     * "additional" input objects that store information such as
+     * mentioned above. These pointers are all to objects whose types
+     * are derived from the current class.
+     *
+     * The format of the additional quantities defined in derived classes
+     * should be the same as for MaterialModel::MaterialModelInputs.
+     */
+    template <int dim>
+    class AdditionalMaterialInputs
+    {
+      public:
+        virtual ~AdditionalMaterialInputs()
+        {}
+
+        virtual void
+        fill (const unsigned int                n_points,
```

Comment:
If we assume that we always want to get as many points as there are quadrature points in the fe_values object anyway, we could remove this parameter (it would be equivalent to `fe_values.n_quadrature_points`). Do you think that would be ok?

Reply:
Yes.